### DISCUSSION TITLE
번역 진행 상황 [CLOSED]

### DISCUSSION THREAD

**Discussion by Gumball12** [vitejs/docs-ko#1] [General]
## 번역 진행 상황

~현재 번역 진행 상황입니다.~

~번역을 진행하고자 하시는 분은 comments를 남겨주세요.~

번역이 마감되었습니다. 감사합니다.

- config
  - [x] [index](https://github.com/vitejs-kr/docs-next/blob/main/config/index.md)
- guide
  - [x] [api-hmr](https://github.com/vitejs-kr/docs-next/blob/main/guide/api-hmr.md)
  - [x] [api-javascript](https://github.com/vitejs-kr/docs-next/blob/main/guide/api-javascript.md)
  - [x] [api-plugin](https://github.com/vitejs-kr/docs-next/blob/main/guide/api-plugin.md)
  - [x] [assets](https://github.com/vitejs-kr/docs-next/blob/main/guide/assets.md)
  - [x] [backend-integration](https://github.com/vitejs-kr/docs-next/blob/main/guide/backend-integration.md)
  - [x] [build](https://github.com/vitejs-kr/docs-next/blob/main/guide/build.md)
  - [x] [comparisons](https://github.com/vitejs-kr/docs-next/blob/main/guide/comparisons.md)
  - [x] [dep-pre-bundling](https://github.com/vitejs-kr/docs-next/blob/main/guide/dep-pre-bundling.md)
  - [x] [env-and-mode](https://github.com/vitejs-kr/docs-next/blob/main/guide/env-and-mode.md)
  - [x] [features](https://github.com/vitejs-kr/docs-next/blob/main/guide/features.md)
  - [x] [index](https://github.com/vitejs-kr/docs-next/blob/main/guide/index.md)
  - [x] [migration](https://github.com/vitejs-kr/docs-next/blob/main/guide/migration.md)
  - [x] [ssr](https://github.com/vitejs-kr/docs-next/blob/main/guide/ssr.md)
  - [x] [static-deploy](https://github.com/vitejs-kr/docs-next/blob/main/guide/static-deploy.md)
  - [x] [why](https://github.com/vitejs-kr/docs-next/blob/main/guide/why.md)
- plugins
  - [x] [index](https://github.com/vitejs-kr/docs-next/blob/main/plugins/index.md)

**Comment by proceane** [comment]
안녕하세요! 번역에 참여해도 될까요?

**Reply by Gumball12** [reply]
넵 언제든지 원하시는 페이지 편하신 분량만큼 해주신 다음 pr 보내주세요 (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧

(서로 겹치지 않게 어떤 페이지 번역해주실건지만 알려주세요!)

**Reply by proceane** [reply]
comparisons 페이지 번역하겠습니다!

**Reply by Gumball12** [reply]
넵 알겠습니다!

**Reply by proceane** [reply]
backend-integration 번역 진행해도 될까요?

**Reply by Gumball12** [reply]
넵 확인했습니다 ㅎㅎ 부탁드리겠습니다

**Comment by choi-me** [comment]
안녕하세요, 저도 번역에 참여하고 싶습니다.

config - index
plugin - index 

우선 번역하고 싶습니다. 알려주시면 감사하겠습니다.

아, 그리고 궁금한점이 있습니다. 살펴보니까, vitejs 영문버전과 불일치하는 메뉴나 부분들이 있는 것 같은데요. 아마 계속 문서가 업데이트 되어서 그런 현상이 있는 것 같아보입니다. 혹시 그런 업데이트들을 어떻게 반영을 하고 계신지 여쭤봐도 될까요?

**Reply by Gumball12** [reply]
안녕하세요. 확인이 늦었습니다.

우선 먼저, 번역에 기여해 주셔서 감사합니다.

(1)
vue와는 다르게 vite의 문서는 `vitejs/vite` 리포지터리의 `docs` 디렉터리에 존재합니다.

처음 설계가 잘못되었던 것이, 큰 변화는 없을 것이라 감히 판단하여 `docs` 디렉터리만 분리해 번역을 진행하고자 했었습니다.


(2)
그러나 말씀해주신 것과 같이 불일치하는 메뉴 및 콘텐츠가 확인되는 바, `vitejs/vite` 리포지터리를 fork 해 본 리포지터리와 merge 하도록 하겠습니다(git history는 소실되지 않도록 하겠습니다).

merge 후, 아래와 같이 branch를 구분해 진행할 계획입니다.

- `main`: `vitejs/vite` 리포지터리와 sync를 위한 브랜치
- `translate`: 한국어 번역을 위한 브랜치
- `gh-pages`: Deploy를 위한 브랜치

Actions를 이용해 일 단위로 docs 디렉터리에 대한 sync를 진행할 계획이며, 변경되는 사항은 `translate` 브랜치에 적용하도록 하겠습니다.

추가적으로, 이 내용은 eterv님께서 번역이 완료된 시점에 진행하도록 하겠습니다.

번역 프로젝트는 경험이 적어 미숙한 부분이 많습니다. 더 좋은 방법이 있는 경우 언제든지 말씀해주세요.


(3)
[`config/index.md`](https://github.com/vitejs-kr/docs-next/blob/main/config/index.md), [`plugin/index.md`](https://github.com/vitejs-kr/docs-next/tree/main/plugins) 두 콘텐츠에 대한 번역을 진행하신다는 것으로 이해됩니다.

진행 상황 업데이트 하도록 하겠습니다. 다만 번역 시 아래 사향을 유의해주세요.

- `#`로 시작하는 Markdown Header 번역 시 url 링크가 정상적으로 연결되지 않습니다.
- 따라서, Header와 연결된 url에 대해 같이 수정을 해주세요.

예를 들어 아래와 같습니다.

```
## Build Options

### build.target

...
```

`vitejs-kr/docs-next` 리포지터리의 [`config/index.md` 406번 째 줄](https://github.com/vitejs-kr/docs-next/blob/main/config/index.md#build-options)은 위와 같이 번역이 되어있지 않았으며, 이를 아래와 같이 번역하는 경우

```
## 빌드 옵션

### build.target

...
```

동일 리포지터리의 `guide/build.md` 34번 째 줄에 있는 [`/config/#build-options` url 링크](https://github.com/vitejs-kr/docs-next/blob/main/guide/build.md#%EB%B9%8C%EB%93%9C-%EC%BB%A4%EC%8A%A4%ED%84%B0%EB%A7%88%EC%9D%B4%EC%A6%88%ED%95%98%EA%B8%B0)는 정상적으로 동작하지 않게 됩니다.

```
빌드와 관련된 커스터마이즈는 [build 설정](/config/#build-options)을 통해 가능합니다.
```

따라서, 해당 url을 정상적으로 Ref할 수 있도록 다음과 같이 업데이트 바랍니다(아래 파일 내용은 `guide/build.md` 파일입니다).

```
빌드와 관련된 커스터마이즈는 [build 설정](/config/#빌드-옵션)을 통해 가능합니다.
```

---

관련 내용 README에 업데이트 하도록 하겠습니다.

추가적으로 여쭈어 보고 싶은 내용이 있으시면 언제든지 Comment 바랍니다.

기여에 감사드립니다.

**Reply by choi-me** [reply]
@Gumball12 

자세한 답변 감사드립니다!

그런데 url 링크 주소를 한국어로 바꾸는 부분은 다소 좋아보이지 않는 것 같습니다.
그래서 이미 중국어로 되어있는 리포지토리를 봤는데요.
[중국어 번역 리포지토리](https://github.com/vitejs/docs-cn)
.vitepress 안에 rewrite-title, custom-anchor 등등의 코드들이 있었습니다.
[중국어 문서의 예제](https://cn.vitejs.dev/config/#async-config)를 확인해 보시면, 영문자로 된 주소를 사용하는 것 같습니다.
코드를 제가 자세히 본 것은 아니고, 대략만 봤는데, 저희 한국어 리포지토리에는 없는 부분들이 있었습니다.
제가 이해한 바가 맞는지 확인해 주시면 감사하겠습니다.

그리고 번역 문제는, 우선 전체 틀을 문제없이 처리해놓고 번역해도 되지 않을까 싶습니다.
텍스트를 바꾸는 것보단 프레임을 우선 잘 구축하는게 먼저이지 않을까 생각이 되네요. 
이참에 vite 문서 최신버전으로 연동시키셔도 될 것 같구요.
다른분이 현재 번역중이지 않다면, 저는 기다릴 수 있습니다.

그럼 답변 부탁드립니다.^^

**Reply by Gumball12** [reply]
@eterv  

말씀해주신 것과 같이 docs-cn 프로젝트는 빌드 시 스크립트를 이용해 anchor의 `id`와 실제 `textContent`를 분리시키는 것을 확인했습니다.

본 리포지터리도 해당 내용 적용해 재구성 및 최신화하도록 하겠습니다.

번역 작업은 그 이후 재개하는 것으로 하겠습니다.

피드백 감사드립니다. 재구성 후 멘션 드리도록 하겠습니다.

**Reply by choi-me** [reply]
@Gumball12 

네, 감사합니다! 그럼, 멘션 주시길 기다리겠습니다.

**Reply by Gumball12** [reply]
@eterv 

안녕하세요. 리포지터리에 대한 Sync CI 구성 및 기존 내용에 대해 최신화 진행하였습니다.

혹 여유가 되신다면, 혹시나 빼먹은 부분이 있는지 확인 부탁드리겠습니다.

- [vitejs/docs-cn](https://github.com/vitejs/docs-cn) 리포지터리를 참고해 Custom Anchor를 추가하여 한글이 아닌 영문으로 Anchor가 생성되게끔 하였습니다.
- 매일 자정, [vitejs/vite](https://github.com/vitejs/vite) 원본과 `docs` 디렉터리에 대해 Sync 작업을 진행합니다. ([sync.yml](https://github.com/vitejs-kr/docs-next/blob/main/.github/workflows/sync.yml))
- 만일 `docs` 디렉터리에 변경사항이 생길 경우, 이를 커밋 단위로 GitHub Issues에 남겨 알리도록 합니다. ([sync script](https://github.com/vitejs-kr/docs-next/blob/main/scripts/sync/index.js))
- 관련 내용을 README에 추가하였습니다.

본 리포지터리에 대한 관심에 감사드립니다.

**Reply by choi-me** [reply]
@Gumball12 

안녕하세요. 감사합니다! 제가 지난 주말에 휴가여서 답을 빨리 못드렸습니다.
일단 해주신 처리는 문제없어 보입니다만, 번역 작업해보면서 혹시 이슈가 생기면 알려드리겠습니다.
다만 제가 휴가후 일이 약간 밀려서 번역 작업은 7월 중순에 처리해도 될까요?

**Reply by Gumball12** [reply]
@eterv 

넵 언제든지 편하실 때, 원하실 때 번역 진행해주시면 됩니다 ㅎㅎ

**Reply by Gumball12** [reply]
@eterv 

Base URL 관련 이슈가 있어 리포지터리 이름을 `docs-next` 에서 `vitejs.github.io` 로 바꾸고자 합니다.

번역 진행 시 참고 바랍니다.

**Reply by choi-me** [reply]
친절한 PR 코멘트에 감사드립니다!! config, plugins 번역에 기여할수 있어 즐거운 경험이었습니다!!
얼른 100% 한글 버전을 Vite 웹에서 볼수 있기를 바래요!!

**Reply by Gumball12** [reply]
넵 노력하겠습니다. 감사합니다! 😊

**Comment by eddie0329** [comment]
안녕하세요! 저도 번역에 참여하고 싶어 이렇게 글을 남깁니다.

`api-hmr` 번역을 하고 싶습니다!

**Reply by Gumball12** [reply]
@eddie0329 
안녕하세요! 번역에 참여하신 것을 환영합니다 🎉
번역을 진행하는 방법은 [Readme](https://github.com/vitejs-kr/vitejs-kr.github.io)를 참고해주세요.
진행하며 의논할 사항은 언제든지 코멘트를 남겨주세요.
감사합니다 😄

**Reply by eddie0329** [reply]
api-hmr.md 부분을 번역하여 PR을 올렸습니다 ! 확인해보시고 이상한 부분 알려주시면 변경하겠습니다 ! 

그럼 좋은 하루 보내세요~~!

**Reply by eddie0329** [reply]
@Gumball12 님 혹시 마지막 남은 `api-javascript` 부분도 제가 번역을 해도 될까요?

**Reply by Gumball12** [reply]
@eddie0329 넵 물론이죠. 부탁드리겠습니다 🙏

**Reply by eddie0329** [reply]
감사합니다! 그럼 진행 후 PR 올려드리겠습니다 ! 🙇🏻‍♂️
